frontend/src/components/ExchangePage.vue

```diff
@@ -303,16 +317,34 @@ export default {
         })
         .finally(() => {
           this.loadingTable = false;
+          this.initialLoad = false;
         });
     },
+    getRequestsWithDebounce: debounce(function() {this.getRequests();}, 1000),
   },
   watch: {
-    pageSize() {
-      this.getRequests();
+    pageSize: {
+      handler() {
+        this.getRequestsWithDebounce();
+      }
+
```

Comment:
Not sure if application should wait for 1 second "debounce" for pagesize and pagenumber changes

Reply:
I could reduce it, but from my testing it seems fast enough. The main reason why the debounce is in there is to prevent multiple page loads when the application starts. 

Since in the creation part of the lifecycle, we are "changing" the page size, page number and search parameters. Since these variables are watched they will trigger a reload for each change. Causing 3 additional requests to the API. 

The debounce methods makes it into 1.

Reply:
Debounce for search parameters seems like a great idea. I am not sure if we should have it for page size and page number I think it could add some not required delays to the User experience. We should discuss this with the whole team to reach consensus.